charts/kong/values.yaml

```diff
@@ -302,6 +302,13 @@ ingressController:
   rbac:
     # Specifies whether RBAC resources should be created
     create: true
+    # Create cluster roles
+    # Cluster roles allow the controller to use resources from all namespaces for Kong configuration
+    # Depending on your K8S cluster configuration, you may not be able to create cluster roles, and can
+    # disable this setting if so. This limits the controller to resources in the namespace it's installed in,
+    # and should be used along with "ingressController.env.watch_namespace: INSTALL_NAMEPSACE"
+    clusterroles:
+      enabled: true
```

Comment:
Can we instead opt for an enum?
```yaml
rbac:
  scope: "cluster" # allows values are "cluster" and "namespace"
```

The reasoning here is that although RBAC is not heavily configurable today, I expect users to be able to tweak RBAC resource heavily in future as the controller is going to get smarter about adjusting it's feature-set based on the RBAC permissions it has. Also, the controller is going to be compatible with multiple API groups:
- core k8s
- configuration.konghq.com
- Service APIs
- Knative
- Possibly some more

The scope of permissions and permissions themselves could be configurable and going down the `clusterroles.enabled` nesting path makes the values.yaml more verbose than it needs to be.

Reply:
Perhaps something like this, allowing us to add the others later?

```
# Settings related to Kubernetes RBAC
rbac:
  scope:
    # namespace-level roles and bindings. should always be enabled
    namespace: true
    # cluster-level roles and bindings. should be disabled in more restricted environments,
    # where Kong administrators do not have access to resources across the K8S cluster,
    # or do not want to grant K8S cluster-level permissions to the Kong cluster
    # when disabled, the chart creates equivalent namespaced roles as best possible
    cluster: true
```

I'd like to both clarify the Kong/K8S distinction (since both have RBAC) and also avoid using a single string to encapsulate all the toggles, since it seems like there may be many in the future. We wouldn't want something like `"clusteralsocontrolleralsoPSPalsoKnative"`.

Setting all to scopes false is equivalent to the currently proposed `rbac.create: false`. We should consider moving `ingressController.rbac.create` to an `rbac.controller` setting for chart 2.0, or otherwise rethink how we split K8S RBAC settings out from their related sections (`ingressController` and `podSecurityPolicy`) if at all.

Kinda torn on whether to allow users to toggle namespace-level roles off--I can't think of much reason to do so, but we kinda allow you to do so currently if I remember how the controller settings work

Reply:
> I'd like to both clarify the Kong/K8S distinction (since both have RBAC) and also avoid using a single string to encapsulate all the toggles, since it seems like there may be many in the future. We wouldn't want something like "clusteralsocontrolleralsoPSPalsoKnative".

That was not my point. My point was rather to have `scope` as an enum of `cluster` and `namespace`. Fine-grained permissions or not part of `scope`. 

Whatever solution we decide should be forward compatible with the following features that _migth_ come in the ingress controller:
- Support for multiple namespaces (current it is either one or all namespaces)
- Support for fine-grained RBAC policies

> Setting all to scopes false is equivalent to the currently proposed rbac.create: false. We should consider moving ingressController.rbac.create to an rbac.controller setting for chart 2.0, or otherwise rethink how we split K8S RBAC settings out from their related sections (ingressController and podSecurityPolicy) if at all.

please track this in #73 

> Kinda torn on whether to allow users to toggle namespace-level roles off--I can't think of much reason to do so, but we kinda allow you to do so currently if I remember how the controller settings work

What do you mean by namespace-level role here? I'm not sure what you mean.

Reply:
That sounds fine, `scope: "cluster"` then.

I just meant a plain Role, as opposed to cluster-scoped ClusterRole.